src/live-watch/live-watch.ts

```diff
@@ -0,0 +1,172 @@
+
+    private roots: LiveWatchNode[] = [];
+    private _context: vscode.ExtensionContext;
+    private continueEvaluate: boolean = true;
+    public activeSession: GDBTargetDebugSession | undefined;
+
+    constructor(private readonly context: vscode.ExtensionContext) {
+        this.roots = this.context.globalState.get<LiveWatchNode[]>(this.STORAGE_KEY) ?? [];
+        this._context = context;
+    }
+
+    getChildren(element?: LiveWatchNode): Promise<LiveWatchNode[]> {
+        if (!element) {
+            return Promise.resolve(this.roots);
+        }
+        return Promise.resolve(element.children ?? []);
+    }
+
+    getTreeItem(element: LiveWatchNode): vscode.TreeItem {
+        const item = new vscode.TreeItem(element.expression, vscode.TreeItemCollapsibleState.None);
+        item.description = element.value || '';
+        item.contextValue = 'expression';
+        return item;
+    }
+
+    getNumberOfNodes(): number {
+        return this.roots.length;
+    }
+
+    activate(tracker: GDBTargetDebugTracker): void {
+        this.addVSCodeCommands();
+        const onDidChangeActiveDebugSession = tracker.onDidChangeActiveDebugSession((session) => {
+            this.activeSession = session;
+            this.refresh();
+            this.continueEvaluate = false;
+        });
+        const onWillStartSession =  tracker.onWillStartSession(session => this.handleOnWillStartSession(session));
+        // Doing a refresh on pausing to ensure we have the latest data
+        const onStopped = tracker.onStopped(() => this.refresh());
+        // Using this event because this is when the threadId is available for evaluations
+        const onStackTrace = tracker.onDidChangeActiveStackItem(() => this.refresh());
+        // Clearing active session on closing the session
+        const onWillStopSession = tracker.onWillStopSession(session => {
+            this.activeSession = session;
+            this.refresh();
+        });
+        this._context.subscriptions.push(onDidChangeActiveDebugSession);
+        this._context.subscriptions.push(onWillStartSession);
+        this._context.subscriptions.push(onStopped);
+        this._context.subscriptions.push(onStackTrace);
+        this._context.subscriptions.push(onWillStopSession);
+    }
+
+    handleOnWillStartSession(session: GDBTargetDebugSession): void {
+        session.refreshTimer.onRefresh(async (refreshSession) => {
+            this.activeSession = refreshSession;
+            this.refresh();
+        });
+    }
+    addVSCodeCommands() {
+        this._context.subscriptions.push(
+            vscode.window.registerTreeDataProvider('cmsis-debugger-view', this)
+        );
+        this._context.subscriptions.push(
+            vscode.commands.registerCommand('cmsis-debugger-view.add', async () => {
+                const expression = await vscode.window.showInputBox({ prompt: 'Expression' });
+                if (!expression) return;
```

Comment:
Please use newlines and curly brackets for such statements. I see this a lot here.
Rather move the logic to handler methods when it gets more than simple one-liners.